ext/mixed/js/text-scanner.js

```diff
@@ -23,11 +23,13 @@
  */
 
 class TextScanner {
-    constructor(node, ignoreElements, ignorePoints) {
+    constructor(node, ignoreElements, ignorePoints, canEnable=null) {
```

Comment:
Rather than adding `canEnable`, might be better to add it as an argument to `setEnabled`. `setEnabled` is only called from `TextScanner.setOptions`, which is called from the subclasses of `TextScanner` (i.e. `Frontend`).

Meaning you can just pass `this.popup.depth <= this.options.scanning.popupNestingMaxDepth && !this._disabledOverride` here:
https://github.com/FooSoft/yomichan/blob/791c51f836c58ea7783acfc077d315c9897f224c/ext/fg/js/frontend.js#L147

And `true` here (or default value):
https://github.com/FooSoft/yomichan/blob/791c51f836c58ea7783acfc077d315c9897f224c/ext/bg/js/search-query-parser.js#L109

This should give better control over when the object is enabled/disabled.

Reply:
https://github.com/FooSoft/yomichan/pull/439/commits/6b47f80f195954f158de44b2971f653c3e23b7b8